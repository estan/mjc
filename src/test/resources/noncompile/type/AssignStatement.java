class AssignStatementTest {
    public static void main(String[] args) {
        int i;
        int[] ia;
        boolean b;
        A cA;
        B cB;

        A = i;  // EXPECTED_VARIABLE_GOT_CLASS
        U = i;  // UNDECLARED_IDENTIFIER

        //i = i; // OK!
        i = ia; // INVALID_ASSIGNMENT
        i = b;  // INVALID_ASSIGNMENT
        i = cA; // INVALID_ASSIGNMENT

        ia = i;  // INVALID_ASSIGNMENT
        //ia = ia;  // OK!
        ia = b;  // INVALID_ASSIGNMENT
        ia = cA; // INVALID_ASSIGNMENT

        cA = i;  // INVALID_ASSIGNMENT
        cA = ia; // INVALID_ASSIGNMENT
        cA = b;  // INVALID_ASSIGNMENT
        //cA = cA; // OK!
        cA = cB; // INVALID_ASSIGNMENT
    }
}
class A {}
class B {}
