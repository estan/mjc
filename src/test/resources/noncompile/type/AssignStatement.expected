EXPECTED_VARIABLE_GOT_CLASS
UNDECLARED_IDENTIFIER
INVALID_ASSIGNMENT
INVALID_ASSIGNMENT
INVALID_ASSIGNMENT
INVALID_ASSIGNMENT
INVALID_ASSIGNMENT
INVALID_ASSIGNMENT
INVALID_ASSIGNMENT
INVALID_ASSIGNMENT
INVALID_ASSIGNMENT
INVALID_ASSIGNMENT
